Content.Shared/Xenoarchaeology/Artifact/XAE/XAEDamageInAreaSystem.cs

```diff
@@ -0,0 +1,37 @@
+using Content.Shared.Damage;
+using Content.Shared.Whitelist;
+using Content.Shared.Xenoarchaeology.Artifact.XAE.Components;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Shared.Xenoarchaeology.Artifact.XAE;
+
+public sealed class XAEDamageInAreaSystem : BaseXAESystem<XAEDamageInAreaComponent>
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+    [Dependency] private readonly EntityWhitelistSystem _whitelistSystem = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    /// <inheritdoc />
+    protected override void OnActivated(Entity<XAEDamageInAreaComponent> ent, ref XenoArtifactNodeActivatedEvent args)
```

Comment:
Just to bikeshed this, I don't think you need this many systems. A partial class that just handles Xenoarch effects makes more sense, then one class per. It's a lot of files for a lot of things that happen in Xenoarch.

You're pretty much entirely going "hey, if this artifact has this particular component and a node activation event happens, fire this function"... it feels like that could be expressed without as much class bloat.

Reply:
I'd like to hear maintainer opinion of that <_<
System per effect is the way it was, i just transferred it.